apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/OperationExtensions.kt

```diff
@@ -134,11 +137,43 @@ fun <D : Operation.Data> Operation<D>.parse(
 }
 
 /**
- * Parses GraphQL operation raw response from the [byteString] with provided [customScalarAdapters] and returns result [Response]
+ * Parses GraphQL operation raw response from [string] with provided [customScalarAdapters] and returns result [Response]
  */
 fun <D : Operation.Data> Operation<D>.parse(
     string: String,
     customScalarAdapters: CustomScalarAdapters = DEFAULT
 ): Response<D> {
   return parse(Buffer().writeUtf8(string), customScalarAdapters)
 }
+
+/**
+ * Parses GraphQL operation raw response from [map] with provided [customScalarAdapters] and returns result [Response]
+ *
+ * @param map: a [Map] representing the response. It typically include a "data" field
+ */
+fun <D : Operation.Data> Operation<D>.parse(
+    map: Map<String, Any?>,
+    customScalarAdapters: CustomScalarAdapters = DEFAULT
+): Response<D> {
+  return MapResponseParser.parse(map, this, customScalarAdapters)
+}
+
+/**
+ * Parses GraphQL operation raw response from [map] with provided [customScalarAdapters] and returns result [Response]
+ *
+ * @param map: a [Map] representing the response. It typically include a "data" field
+ */
+fun <D : Operation.Data, M: Map<String, Any?>> Operation<D>.parseData(
+    map: M,
+    customScalarAdapters: CustomScalarAdapters = DEFAULT,
+    valueResolver: ValueResolver<M>
+): D {
```

Comment:
should we keep just one API with default argument for `valueResolver`?

Reply:
The above parses a response, this one only parses `data` so we can't really merge them. `parseData` is needed because when reading from cache, we don't have a proper `Response`